spec/factories/fact_checkpoints.rb

```diff
@@ -0,0 +1,8 @@
+FactoryGirl.define do
+  factory :fact_checkpoint do
+    simulation JSON.generate sample: {values: [1], errors: []}, stats: {mean: 1, stdev: 0, length: 1}
```

Comment:
I added validations to that affect to the `Fact` model (as opposed to in the `fact_checkpoints` model). Let me know if I misunderstood your concern.